presto-native-execution/presto_cpp/main/operators/PartitionAndSerialize.cpp

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "presto_cpp/main/operators/PartitionAndSerialize.h"
+#include "velox/connectors/hive/HivePartitionFunction.h"
+#include "velox/exec/HashPartitionFunction.h"
+#include "velox/expression/Expr.h"
+
+#include "velox/row/UnsafeRowDynamicSerializer.h"
+
+using namespace facebook::velox::exec;
+using namespace facebook::velox;
+
+namespace facebook::presto::operators {
+
+void PartitionAndSerializeNode::addDetails(std::stringstream& stream) const {
+  stream << "(";
+  for (auto i = 0; i < keys_.size(); ++i) {
+    const auto& expr = keys_[i];
+    if (i > 0) {
+      stream << ", ";
+    }
+    if (auto field =
+            std::dynamic_pointer_cast<const core::FieldAccessTypedExpr>(expr)) {
+      stream << field->name();
+    } else if (
+        auto constant =
+            std::dynamic_pointer_cast<const core::ConstantTypedExpr>(expr)) {
+      stream << constant->toString();
+    } else {
+      stream << expr->toString();
+    }
+  }
+  stream << ") " << numPartitions_;
+}
+
+namespace {
+
+class PartitionAndSerializeOperator : public Operator {
+ public:
+  PartitionAndSerializeOperator(
+      int32_t operatorId,
+      DriverCtx* FOLLY_NONNULL ctx,
+      const std::shared_ptr<const PartitionAndSerializeNode>& planNode)
+      : Operator(
+            ctx,
+            planNode->outputType(),
+            operatorId,
+            planNode->id(),
+            "PartitionAndSerialize") {
+    auto inputType = planNode->sources()[0]->outputType();
+    auto keyChannels = toChannels(inputType, planNode->keys());
+
+    // Initializing hive partition function
```

Comment:
Comments should be full sentences, e.g. start with a capital letter and end with a period. Please, fix throughout the PR.